File: 0234-palindrome-linked-list/0234-palindrome-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        string s;
        while(head!=NULL){
            s.push_back(head->val);
            head=head->next;
        }
        for(int i=0;i<s.length();i++){
            if(s.at(i)!=s.at(s.length()-i-1)){
                return 0;
            }
        }
        return 1;
    }
};

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 * int val;
 * ListNode next;
 * ListNode() {}
 * ListNode(int val) { this.val = val; }
 * ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        Stack<Integer> s = new Stack<>();
        if(head.next==null){
            return true;
        }
        while (head != null) {
            if (!s.isEmpty() && s.peek() == head.val) {
                s.pop();
            } else
                s.push(head.val);
            head = head.next;
        }
        if (s.isEmpty()) {
            return true;
        }
        return false;
    }
}